sycl/test-e2e/Matrix/Inputs/joint_matrix_out_bounds_impl.hpp

```diff
@@ -54,27 +48,41 @@ void matrix_multiply(T1 *C, T2 *A, T2 *B, queue q) {
            const auto sg_starty = global_idy - spmd_item.get_local_id(1);
 
            sub_group sg = spmd_item.get_sub_group();
-           joint_matrix<sub_group, bfloat16, use::a, TM, TK, layout::row_major>
-               sub_a;
-
-           // For B, since current implementation does not support non-packed
-           // layout, users need to specify the packed_b layout.
-           joint_matrix<sub_group, bfloat16, use::b, TK, TN, B_layout> sub_b;
-           joint_matrix<sub_group, float, use::accumulator, TM, TN> sub_c;
-           // bounds-checked load where width and height are added
+           joint_matrix<sub_group, T2, use::a, TM, TK, A_layout> sub_a;
+           joint_matrix<sub_group, T2, use::b, TK, TN, B_layout> sub_b;
+           joint_matrix<sub_group, T1, use::accumulator, TM, TN> sub_c;
+
+           // bounds-checked fill where width and height are added
            ext::intel::experimental::matrix::joint_matrix_fill_checked(
```

Comment:
We should also add load of accumulator C both row and column major. It does not exist today in the tests. May be leave the fill_checked for only one of the test instances.

Reply:
Yes, we should, but it goes beyond of what I planned for this PR.